### ISSUE TITLE
List patterns: duplicate errors when no suitable range indexer

### ISSUE THREAD

**Issue by SomeAnonDev** [dotnet/roslyn#59529]
**Version Used**: main (27 Jan 2022, https://github.com/dotnet/roslyn/commit/3b32fc36ec4c2a81457d90f965c89c5b6fc37082)

**Steps to Reproduce**:

```csharp
using System;

{
    _ = new Missing() is [..var x];
}
{
    _ = new Inaccessible() is [..var x];
}

public struct Missing
{
    public int Length => 42;
}

public struct Inaccessible
{
    public int Length => 42;
    private Inaccessible this[Range r] => default;
}
```

**Expected Behavior**: each of the two errors is reported without duplicates

**Actual Behavior**:
```
error CS0021: Cannot apply indexing with [] to an expression of type 'Missing'
error CS0021: Cannot apply indexing with [] to an expression of type 'Missing'
error CS0122: 'Inaccessible.this[Range]' is inaccessible due to its protection level
error CS0122: 'Inaccessible.this[Range]' is inaccessible due to its protection level
```
https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKAIG8CACc0gfVIF5SYBTAd1IFkoBnD2AcwAoAlKU6kA2gDpxANwgAnUgA8AugG4CAX2JkK1OoxYBJGBADGJhlyjAANg0HCOYyTPnK1+TYXzoAzKQ4ALrIAriYBbJzcMDxa+BSkvsIw4QAyDNEBABa0AHykACyY7p4EiYEhYaRGpuaWNgyx8Ymwqek8WbkFRdrkAA6yUDIBDFXGZhbc9aRZnKIAShDRI7JKnQgMAGYQwdYBxUA=

![image](https://user-images.githubusercontent.com/6377116/153883056-b0d2370c-e716-4956-9143-872a74587819.png)

P.S. is the entire list pattern really supposed to be highlighted here when it's only the slice pattern that requires a range indexer?
![image](https://user-images.githubusercontent.com/6377116/153883438-0d70191d-def3-4e4c-8402-259bada6d021.png)

**Comment by SomeAnonDev** [comment]
Okay, sure, there's no suitable ``System.Index`` indexer either, and that could be why the errors appear duplicated (and also why the entire list pattern is highlighted).
However, the error message refers to ``this[Range]`` in both instances.

**Comment by jcouv** [comment]
FYI @alrz
